app/javascript/kramdown.jsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+
+/**
+ * Escapes user input for storage in a Kramdown extension attribute.
+ * Note that this _does not_ do sanitization--that'll be the job of
+ * whatever's rendering the Markdown to HTML.
+ * @param {any} input
+ * @returns {string}
+ */
+function escapeForExtensionAttribute(input) {
+  return String(input ?? "").replace(/"/g, '\\"');
+}
+
+/**
+ * @param {any} value
+ * @returns {string}
+ */
+function unescapeExtensionAttribute(value) {
+  return String(value ?? "").replace(/\\"/g, '"');
+}
+
+/**
+ * Generates a regular expression that will match a Kramdown extension, e.g.:
+ *
+ * ```
+ * {::some_tag attr="value" otherAttr="value" /}
+ * ```
+ *
+ * @param {name} {string}
+ * @param {attrNames} {string[]} Names of attributes expected (in order)
+ * @returns {RegExp} A regular expression that will match a kramdown extension tag.
+ */
+function makeExtensionRegex(name, attrNames) {
+  // NOTE: This pattern is quite fragile and depends on exact ordering of attributes.
```

Comment:
[question] I'm not terribly worried about the fragility of the pattern since we own the code that builds and parses the extension. However, are there any guardrails we can put in place to warn a developer of issues if/when they add a new field to an existing editor? Maybe this is pointing to a need for javascript unit tests?

Reply:
Yeah, I've been trying not to think about bringing in a JS unit testing framework 😄. It is probably worth a few hours to see how it works out. I will create a story for that.